tests/e2e/envfuncsext/nebulacluster-ready-func.go

```diff
@@ -173,3 +194,41 @@ func isComponentStatusExpected(
 
 	return true
 }
+
+func isComponentResourceExpected(ctx context.Context, cfg *envconf.Config, component appsv1alpha1.NebulaClusterComponent) bool {
+	sts := &appsv1.StatefulSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      component.GetName(),
+			Namespace: component.GetNamespace(),
+		},
+	}
+
+	if err := cfg.Client().Resources().Get(ctx, sts.Name, sts.Namespace, sts); err != nil {
```

Comment:
We have already obtained it, why not check more fields, such as version, replicas , etc.